src/spaces/gradedspace.jl

```diff
@@ -92,10 +92,9 @@ Base.hash(V::GradedSpace, h::UInt) = hash(V.dual, hash(V.dims, h))
 field(::Type{<:GradedSpace}) = ℂ
 InnerProductStyle(::Type{<:GradedSpace}) = EuclideanInnerProduct()
 function dim(V::GradedSpace)
-    return reduce(
-        +, dim(V, c) * dim(c) for c in sectors(V);
-        init = zero(dim(unit(sectortype(V))))
-    )
+    T = Base.promote_op(*, Int, real(sectorscalartype(sectortype(V))))
+    return reduce(+, dim(V, c) * dim(c) for c in sectors(V);
+                  init=zero(T))
```

Comment:
I don't know if we ended up trying this, but I am aware of `sum` having some special handling of `init` values.
I think we could maybe avoid the `promote_op` machinery by using something that is a little more friendly on inference: (double check type stability though!)

```suggestion
    blockdim(c) = dim(V, c) * dim(c)
    return sum(blockdim, sectors(V))
```

Reply:
`sum` also supports the `init` kwarg. But then indeed you should recompute it. Though I am not sure how `real(sectorscalartype(sectortype(V)))` tells you whether the quantum dimensions are real or integer?

Reply:
It doesn't. I was thinking at the level of the F-symbols, but for certain sectors (in particular `SU2Irrep`) the `dim(::Sector)` is specialised to return integers, while the F-symbols are floats.

Reply:
Does this not work and produce stable results?
```julia
if isempty(sectors(V))
    return dim(first(allunits(sectortype(V)))) * 0
else
    blockdim(c) = dim(V, c) * dim(c)
    return sum(blockdim, sectors(V))
end
```

I think the trade-off is whether we want to simply always compute this `init`, and if not whether we think `Base.promote_op` is acceptable